app/Main.hs

```diff
@@ -1,153 +1,6 @@
 module Main where
 
-import qualified Actions.Add as Actions
-import qualified Actions.Config as Actions
-import qualified Actions.CreateNixFile as Actions
-import qualified Actions.Direnv as Actions
-import qualified Actions.Pin as Actions
-import qualified Actions.Remove as Actions
-import qualified Actions.Search as Actions
-import qualified Actions.Shell as Actions
-import qualified Actions.Tags as Actions
-import Data.Coerce
-import Data.Foldable (traverse_)
-import Data.Maybe (fromMaybe)
-import Options
-import Types.Add
-import Types.Config
-import Types.Shell
-import Types.Tags
-
-nixMateConfig :: String
-nixMateConfig = "./nix-mate.json"
-
-direnvConfig :: String
-direnvConfig = "./.envrc"
+import qualified NixMate.Main as NM
```

Comment:
Having all this crap in `app` made it hard to typecheck and was pointless, this moves it.